src/Configuration.php

```diff
@@ -353,13 +360,29 @@ public function setAppType($type)
     }
 
     /**
+     * Set the fallback application type.
+     *
+     * This is should be used only by libraries to set an fallback app type.
+     *
+     * @param string|null $type the fallback type
+     *
+     * @return $this
+     */
+    public function setFallbackType($type)
+    {
+        $this->fallbackType = $type;
+
+        return $this;
+    }
+
+    /**
      * Get the application data.
      *
      * @return array
      */
     public function getAppData()
     {
-        return array_merge(['type' => php_sapi_name(), 'releaseStage' => 'production'], array_filter($this->appData));
+        return array_merge(['type' => $this->fallbackType ?: php_sapi_name(), 'releaseStage' => 'production'], array_filter($this->appData));
```

Comment:
That would seem good - that way all the defaults are centralised and we dont have to worry about merging things in the right order so much. We probably still need the fallbackType from what you were saying because we dont know during construction what app we are running in all the time, but it limits the complexity a bit I think